lib/util/state.js

```diff
@@ -25,32 +25,136 @@ export function getTimestamp (time = moment()) {
+  if (arr.length > 2) {
+    const last = arr.pop()
+    return arr.join(', ') + ' and ' + last
+  } else {
+    return arr.join(' and ')
+  }
+}
+
+/**
+ * Generates a list of issues with a list of OTP responses. For batch routing
+ * this could be an array of multiple responses, but for standard itinerary
+ * routing it will be at most an array with one response.
+ *
  * @param {Object} otpState the OTP state object
- * @return {Array}      array of OTP plan responses that contain errors
+ * @return {Array} An array of objects describing the errors seen. These will at
+ *   a minimum contain the `msg` object describing the error, but could also
+ *   include an `id` and `modes` key for trip planning errors or `network` for
+ *   errors with vehicle rentals
  */
 export function getResponsesWithErrors (otpState) {
   const search = getActiveSearch(otpState)
-  const errorResponses = []
+  const rentalFeedErrors = []
+  const networksWithFeedWideErrors = {}
+  const tripPlanningErrors = []
   const response = !search ? null : search.response
   if (response) {
+    const showModes = response.length > 1
     response.forEach(res => {
-      if (res && res.error) errorResponses.push(res)
+      if (res) {
+        if (res.error) {
+          let msg = res.error.msg
+          // include the modes if doing batch routing
+          if (showModes) {
+            const mode = humanReadableMode(res.requestParameters.mode)
+            msg = `No trip found for ${mode}. ${msg.replace(/^No trip found. /, '')}`
+          }
+          tripPlanningErrors.push({
+            id: res.error.id,
+            modes: res.requestParameters.mode.split(','),
+            msg
+          })
+        }
+        const feedWideRentalErrors = getFeedWideRentalErrors(res)
+        feedWideRentalErrors.forEach(networkWithFeedWideError => {
+          const { network } = networkWithFeedWideError
+          if (!networksWithFeedWideErrors[network]) {
+            rentalFeedErrors.push({
+              msg: `The ${network} network is unavailable at this moment.`,
+              network
+            })
+            networksWithFeedWideErrors[network] = true
+          }
+        })
+      }
     })
   }
-  return errorResponses
+
+  // generate list so rental errors show up first
+  return [...rentalFeedErrors, ...tripPlanningErrors]
 }
 
 /**
- * Gets the active error (first OTP plan response that contains an error). Only
- * one OTP response is expected in itinerary routing, but if batch routing is
- * enabled, multiple responses with errors should be handled.
+ * Gets the active error (first OTP plan response that contains an error). This
+ * should only be used with itinerary routing, not batch routing.
  */
 export function getActiveError (otpState) {
-  const errorResponse = getResponsesWithErrors(otpState)[0]
-  if (errorResponse && errorResponse.error) return errorResponse.error
+  return getResponsesWithErrors(otpState)[0]
```

Comment:
Will this now show just the feedWide rental error if the list of errors has a rental error + OTP response error? If so, is that the desired behavior?

Reply:
Changed so it always gets the last error in 8528d89154ec5e471d8ac40fc66ce4e35d974261